pkg/security/filters/middlewares/authrization.go

```diff
@@ -41,6 +41,17 @@ func (m *Authorization) Run(request *web.Request, next web.Handler) (*web.Respon
 	switch decision {
 	case http.Allow:
 		ctx := request.Context()
+		userContext, found := web.UserFromContext(ctx)
+		if !found {
+			return nil, security.UnauthorizedHTTPError("authorization failed due to missing authentication")
+		}
+		if accessLevel == web.DefaultAccess {
+			return nil, security.UnauthorizedHTTPError("authorization failed due to missing access level")
```

Comment:
It's strange to get `allowed` and then still get `denied` because of access level.
Also, this error message is somewhat confusing. I have default access, it's not missing.